### PR TITLE
Replace `StraightOISOptimisationProblem` in new optimisation feature branch

### LINKED ISSUE #2231: Update `StraightOISOptimisationProblem` for new optimisation interface

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

Closes #2231 

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: tests/builders/test_coil_supports.py [hunk 1]
```diff
@@ -308,6 +308,12 @@ def test_that_the_right_number_of_OIS_are_made(self, koz, n_ois):
         designer = StraightOISDesigner(self.params, {}, self.tf_xz_face, koz)
         ois_wires = designer.run()
         assert len(ois_wires) == n_ois
+        # regression test:
+        # print([w.length for w in ois_wires])
+        # previous: [7.938291929677863, 8.14804252516768, 3.9195699898575236]
+        #           [7.938291929677863, 8.149554771020297]
+        # now:      [7.938291929677863, 8.14804252516768, 3.9195699898575236]
+        #           [7.938291929677863, 8.149554771020297]
```
Comment (hsaunders1904) [inline tests/builders/test_coil_supports.py:316]: Do we actually want to test this, or can we remove the comment?

**Review by hsaunders1904 — APPROVED** [review]
Looks good to me - although we may want to wait for #2251 and make updates before merging

**Review by hsaunders1904 — APPROVED** [review]
I'm happy with this. #2251 is merged now, do you want to update to use that? I can do it if you're strapped for time. Another option is to merge this and then update to use the interface later

**Review by je-cook — APPROVED** [review]
Cool, looks good to me

**Pull request merged**